### PR TITLE
Fix for Plotly chart not rendering 

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves issue under experiment tracking where Plotly graph is not rendering.

Screenshot 2024-01-08 at 9 55 32 a m

## Development notes

Considering the nested nature of the `trackingData`, it could be a data immutability issue causing the problem. To address this, I created a deep copy of the `trackingData` using `structuredClone` and passed it to the `<PlotlyChart>` component.
  

## Checklist

- [X] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by rashidakanchwala — APPROVED** [review]
Amazing!! This has been a mind-boggling issue. We even faced this last year when building out metric plots. So glad it is solved. 

Thanks a lot @jitu5

File: src/components/experiment-tracking/run-dataset/run-dataset.test.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import RunDataset from '.';
 import { runs, trackingData } from '../../experiment-wrapper/mock-data';
 import JSONObject from '../../json-object';
 import { shallow, mount } from 'enzyme';
+import 'core-js/stable/structured-clone';
```
Comment (ravi-kumar-pilla) [inline src/components/experiment-tracking/run-dataset/run-dataset.test.js:6]: Is this required ?

File: src/components/experiment-tracking/run-dataset/run-dataset.js [hunk 2]
```diff
@@ -77,7 +77,12 @@ const RunDataset = ({
   trackingData,
   theme,
 }) => {
-  if (!trackingData) {
+  const clonedTrackingData = useMemo(
+    () => structuredClone(trackingData),
```
Comment (ravi-kumar-pilla) [inline src/components/experiment-tracking/run-dataset/run-dataset.js:81]: Awesome !!

**Review by ravi-kumar-pilla — APPROVED** [review]
Looking great and works well !! Nice to have this in the release note. Thank you @jitu5

**Pull request merged**